# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/MIR.h
# Commit: 89a54069f124
# Full Hash: 89a54069f124b175a0069affaaa1f55ff83214de
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2020-06-16 15:49:59
# Regressor Bug: 1645353
# File Overlap Count: 1
# Description:
#   Bug 1645353 part 2 - Use the template object for scripted constructor calls. r=evilpie,iain
#   
#   Also emit the .prototype guard directly in CacheIR. This way we don't need any of
#   the code IonBuilder has for this in Warp. Only do this if WarpBuilder is enabled
#   because Baseline doesn't need it and this way we avoid potential Ion perf
# ==============================================================================

diff -r 83569838f633 -r 89a54069f124 js/src/jit/MIR.h
--- a/js/src/jit/MIR.h	Sun Jun 14 10:32:32 2020 +0000
+++ b/js/src/jit/MIR.h	Mon Jun 15 15:46:48 2020 +0000
@@ -3544,8 +3544,10 @@
       : MUnaryInstruction(classOpcode, templateConst),
         initialHeap_(initialHeap) {
     setResultType(MIRType::Object);
-    setResultTypeSet(
-        MakeSingletonTypeSet(alloc, constraints, templateObject()));
+    if (!JitOptions.warpBuilder) {
+      setResultTypeSet(
+          MakeSingletonTypeSet(alloc, constraints, templateObject()));
+    }
   }
 
  public: